lib/src/memory/datastore.rs

```diff
@@ -147,58 +147,40 @@ impl InternalMemoryDatastore {
         }
     }
 
-    fn delete_vertices(&mut self, vertices: Vec<Uuid>) {
-        for vertex_id in vertices {
-            self.vertices.remove(&vertex_id);
+    fn delete_vertex(&mut self, vertex_id: Uuid) {
+        self.vertices.remove(&vertex_id);
 
-            let mut deletable_vertex_properties: Vec<(Uuid, String)> = Vec::new();
+        let mut deletable_vertex_properties: Vec<(Uuid, String)> = Vec::new();
 
-            for (property_key, _) in self.vertex_properties.range((vertex_id, "".to_string())..) {
-                let &(ref property_vertex_id, _) = property_key;
+        for (property_key, _) in self.vertex_properties.range((vertex_id, "".to_string())..) {
+            let &(ref property_vertex_id, _) = property_key;
 
-                if &vertex_id != property_vertex_id {
-                    break;
-                }
-
-                deletable_vertex_properties.push(property_key.clone());
-            }
-
-            for property_key in deletable_vertex_properties {
-                self.vertex_properties.remove(&property_key);
+            if &vertex_id != property_vertex_id {
+                break;
             }
 
-            let mut deletable_edges: Vec<EdgeKey> = Vec::new();
+            self.vertex_properties.remove(&property_key);
```

Comment:
This is iterating through `self.vertex_properties` and removing elements at the same time, which should fail AFAIK